### PR TITLE
Integrate Tutorial creation with backend

### PR DESCRIPTION
**Pull request by shivareddy6**
## Description
Integrate the tutorial creation process with backend completely and implement new data base model
model link: https://drive.google.com/file/d/171poQbBFT8-sjec5y_tcTExo3VRyGv8S/view?usp=sharing

## Related Issue
https://github.com/scorelab/Codelabz/issues/845

Commit-1:
- Refactored editor instances for individual steps
- Fixed preview mode (now it shows respective step's content)

https://github.com/scorelab/Codelabz/assets/74174769/7dd750e9-8895-4dc5-b452-7cb987d36622

### CODE REVIEW

**Comment by shivareddy6** [comment]
Second commit:
- Connected editor to backend (cloud firestore)
- Implemented new data base design for tutorials

Video:

https://github.com/scorelab/Codelabz/assets/74174769/d0f2a62b-fcde-4a07-afbe-7683d961ac01

File: src/store/actions/tutorialsActions.js [hunk 1]
```diff
@@ -130,44 +129,36 @@ export const createTutorial =
...
-          updatedAt: firestore.FieldValue.serverTimestamp(),
-        });
+            created_by,
+            editors: [created_by],
+            owner,
+            summary,
+            title,
+            featured_image: "",
+            icon: "",
+            url: "",
```
Comment (Sarfraz-droid) [inline src/store/actions/tutorialsActions.js:146]: Greatt work so far!!
Just a doublt, 
Is there any way to ensure that this post is draft/archived/published ? If not, then we can add these attributes to the set.

Else how are are handling draft & published posts right now

Comment (shivareddy6) [reply src/store/actions/tutorialsActions.js:148]: I somehow missed it. Thanks for pointing out!
I implemented it in the following commit

**Review by Sarfraz-droid — APPROVED** [review]

**Comment by Sarfraz-droid** [comment]
LGTM

**Review by ABHISHEK-PANDEY2 — APPROVED** [review]

**Review by Maahi10001 — APPROVED** [review]

**Pull request merged**